src/rust/engine/fs/store/src/immutable_inputs.rs

```diff
@@ -36,6 +37,7 @@ pub struct ImmutableInputs(Arc<Inner>);
 
 impl ImmutableInputs {
   pub fn new(store: Store, base: &Path) -> Result<Self, String> {
+    create_dir_all(base).map_err(|e| format!("Failed to create base for immutable inputs: {e}"))?;
```

Comment:
This function is already doing blocking filesystem work... but just checking: should the IO here be async (and thus `new` too)? As in, is this called in an async context?

Reply:
Yeah I just used fs because tempbuilder is already synchronous